python/az/aro/azext_aro/_aad.py

```diff
@@ -58,6 +58,13 @@ def get_service_principal(self, app_id):
             return sps[0]
         return None
 
+    def get_application_by_client_id(self, client_id):
```

Comment:
why not use self.client.applications.get ?

Reply:
because get required `application_object_id` as required parameters to be provided and this would mean I have to get it from somewhere. By doing list and filter I can workaround this and just have clientID